File: Easy/You and your books.py

class Solution:
    #Your task is to complete this function
    #Function should return an integer
    #a - list/array containing height of stack's respectively
    def max_Books(self, n, k, arr):
        #code here
        supp = [ 0 ] * (n + 1)
        
        for i in range(n):
            if arr[i] <= k:
                supp[i + 1] = supp[i] + arr[i]
            else:
                supp[i + 1] = 0
        
        ans = -1
        
        for i in range(n + 1):
            #print(supp[i], end = " ")
            ans = max(ans, supp[i])
        
        #print()
        return ans
                


#{ 
 # Driver Code Starts
if __name__ == '__main__':
    t = int(input())
    for i in range(t):
        temp = list(map(int, input().strip().split()))
        n = temp[0]
        k = temp[1]
        arr = list(map(int, input().strip().split()))
        ob = Solution()
        print(ob.max_Books(n, k, arr))
# Contributed by:Harshit Sidhwa

# } Driver Code Ends

File: Easy/You and your books.cpp
//{ Driver Code Starts
#include <iostream>
#include<algorithm>
#include<cstdio>
//#include<Windows.h> 
using namespace std;  

// } Driver Code Ends

class Solution{
    public:
    /*You are requried to complete this method */
    int max_Books(int a[], int n, int k)
    {
        // Your code here
        int somma=0;
        int ans=0;
        for(int i=0;i<n;i++)
        {
            if(a[i]<=k)
            {
                somma+=a[i];
                ans=max(ans,somma);
            }
            else somma=0;
        }
       return ans;
    }
};

//{ Driver Code Starts.
int main()
{
  int t;
  cin>>t;
  while(t--)
  {
  	int n,k;
  	cin>>n>>k;
  	int ar[n];
  	for(int i=0;i<n;i++)
  	{
  		cin>>ar[i];
  	}
  	Solution ob;
  	cout<<ob.max_Books(ar,n,k)<<endl;
        
  }    
  return 0;
}


// } Driver Code Ends